src/Component.php

```diff
@@ -2,15 +2,75 @@
 
 declare(strict_types=1);
 
-namespace MyComponent;
+namespace Keboola\SnowflakeParametersExtractor;
 
 use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvFile;
+use Keboola\Db\Import\Exception as SnowflakeException;
+use Keboola\Db\Import\Snowflake\Connection;
 
 class Component extends BaseComponent
 {
     public function run(): void
     {
-        // @TODO implement
+        $parametersFetcher = new ParametersFetcher($this->createSnowflakeConnection());
+        $parameters = $this->addHostToParameters(
+            $parametersFetcher->fetchAccountParameters()
+        );
+
+        $csvFile = new CsvFile($this->getAccountParametersTableDestination());
+        foreach ($parameters as $parameter) {
+            $csvFile->writeRow($parameter);
+        }
+
+        $this->getManifestManager()->writeTableManifestFromArray(
+            $this->getAccountParametersTableDestination(),
+            [
+                'primary_key' => [
+                    'host',
+                    'key',
+                ],
+                'columns' => array_keys($parameters[0]),
+            ]
+        );
+    }
+
+    private function addHostToParameters(array $parameters): array
```

Comment:
Je škoda, že tohle není někde, kde by to bylo otestované. A nemuselo by to tahat `$host` z configu, ale mohlo by ho to dostat v parametru. 

Zpětně vidím chybu v tom, jak jsem nastavil tu "BaseComponent", protože teď se ukazuje, že je netriviální ji nainicializovat a tím pádem všechny metody v ní typicky zůstanou neotestované :( A lidi mají většinou tendenci do ní dát business logiku. Přitom je to spíš takový controller - měl by jen vzít vstupy (config + manifest) a pak spustit nějaký "business kód" a na konci zapsat výsledky (/out/* + manifesty).

Reply:
me to prijde otestovane tim funkcionalnim testem, a davat to nekam jinam - proc? cela ta trida ma 80 radku i s chlupama, nechtel bych psat jak v jave 4 interfacy a 5 factory trid

Reply:
@tomasfejfar neřekl bych že je chyba v tom jak je to navržené. Myslím že je z toho zřejmé že je to myšleno tak jak popisuješ. Zároveň jsem to nechěl nějak overengineerovat když jde o komponentu volající jeden dotaz do SNFLK:) 

Zároveň ale rozumím tomu že kdyby přišly další požadavky na tu komponentu a pokračovalo se v tom jak je to teď tak to začne být problém.

Já bych to teď releasnul a pak se k tomu schválně vrátil a zkusil poslat PR jako takové cvičení jakmile budu mít chvilku.

Reply:
@odinuv jo, je pravda, že tady v tomhle případě je to maličké, tak máš pravdu :) 

Moje zkušenost byla vždycky, že když se podcenil začátek, tak pak když se měla přidat jedna další věc, tak už to nebylo "hele, ok, pojďme to teď přepsat", ale "je to na 15 minut, tak to tam nějak přidáme a až to bude moc, tak to RF" (x10). Stačí se podívat na kbc, kde i když je to ve slušném stavu, tak na spoustě míst je vidět, že někdo tam udělal nějaký PoC, kus zkopíroval, přidal jeden if a parametr do API a bylo to. Navíc vím, že i já sám tohle dělám :) Tak mám tendenci to tlačit co nejdřív správně ;)

Reply:
ok, já teď použiju ten extractor na to co potřebuji. A pak to tady zkusím zrefaktorovat, mám tam pár otazníků tak jsem zvědavý.

Reply:
Super, budu rád, když tam odhalíš nějaké problémy. Externí vývojáři se s nima neozvou.